spikeextractors/extractors/nwbextractors/nwbextractors.py

```diff
@@ -686,61 +688,66 @@ def add_electrical_series(
-                   and isinstance(metadata['Ecephys']['LFPElectricalSeries'], dict)), \
-               ("Expected either metadata['Ecephys']['ElectricalSeries'] or "
-                "metadata['Ecephys']['LFPElectricalSeries'] to be a dictionary!")
-
-        if not write_as_lfp:
-            es_name = metadata['Ecephys']['ElectricalSeries'].get('name', raw_defaults['name'])
-            assert es_name not in nwbfile.acquisition, \
-                f"Raw ElectricalSeries '{es_name}' is already written in the NWBFile!"
-        else:
-            es_name = metadata['Ecephys']['LFPElectricalSeries'].get('name', lfp_defaults['name'])
+        assert write_as in ['raw', 'processed', 'lfp'], \
+            f"'write_as' should be 'raw', 'processed' or 'lfp', but intead received value {write_as}"
+        if write_as == 'raw':
+            eseries_kwargs = dict(
+                name="ElectricalSeries_raw",
+                description="Raw acquired data",
+                comments="Generated from SpikeInterface::NwbRecordingExtractor"
+            )
+        elif write_as == 'processed':
+            eseries_kwargs = dict(
+                name="ElectricalSeries_processed",
+                description="Processed data",
+                comments="Generated from SpikeInterface::NwbRecordingExtractor"
+            )
+        elif write_as == 'lfp':
+            eseries_kwargs = dict(
+                name="ElectricalSeries_lfp",
+                description="Processed data - LFP",
+                comments="Generated from SpikeInterface::NwbRecordingExtractor"
+            )
+
+        # If user passed metadata info, overwrite defaults
+        if es_key is not None and metadata is not None:
+            assert es_key in metadata.get('Ecephys', dict()).keys(), f"Metadata dictionary does not contain key '{es_key}'"
+            eseries_kwargs.update(metadata['Ecephys'][es_key])
+
+        # Check for existing names and processing modules in nwbfile
+        if write_as == 'raw':
+            assert eseries_kwargs['name'] not in nwbfile.acquisition, f"Raw ElectricalSeries '{es_name}' is already written in the NWBFile!"
+        elif write_as == 'processed':
             ecephys_mod = check_module(
-                nwbfile,
-                'ecephys',
-                "Intermediate data from extracellular electrophysiology recordings, e.g., LFP."
+                nwbfile=nwbfile,
+                name='ecephys',
+                description="Intermediate data from extracellular electrophysiology recordings, e.g., LFP."
+            )
+            if 'Processed' not in ecephys_mod.data_interfaces:
+                ecephys_mod.add_data_interface(FilteredEphys(name='Processed'))
+        elif write_as == 'lfp':
+            ecephys_mod = check_module(
+                nwbfile=nwbfile,
+                name='ecephys',
+                description="Intermediate data from extracellular electrophysiology recordings, e.g., LFP."
```

Comment:
Might as well combine these blocks with the default population above

Reply:
I share the feeling, but if you look at it, we should check the possibility of the `name` coming from the user's metadata already existing in the file, and we can only test it after updating

Reply:
That applies to https://github.com/luiztauffer/spikeextractors/blob/nwb_write_options/spikeextractors/extractors/nwbextractors/nwbextractors.py#L727-L728, sure, but not to the others. From the earliest point that we check the `write_as` values, we know if we need to check the processing modules. Saves us a few lines moving that stuff above.

Reply:
moved, but then I noticed we were not testing for existing name when lfp/processed. added that as well